### PR TITLE
All posts redesign

### PR DESCRIPTION
**Pull request by oetherington**
[Figma (for CEA)](https://www.figma.com/file/OD4vJlLuWriHqG1WJXY4LV/Q1.1-2023?node-id=801%3A37266&t=rBfNfKc8wLWU1fuP-0).

image

We (well, Ollie) made some judgement calls about the way to display Topic wiki edits.

We removed the horizontal rule below the title (All posts) when you open the settings:

image

As well as some judgement calls about the settings.

@agnesstenlund you may want to look at the font sizes of the TimeBlocks and the subsections.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204185280213984) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by agnesstenlund** [comment]
this is beautiful 🥇 agree with all judgement calls

I only had two tiny remarks–pushed a commit to fix those

File: packages/lesswrong/components/posts/PostsTimeBlock.tsx [hunk 1]
```diff
@@ -36,8 +48,14 @@ const styles = (theme: ThemeType): JssStyles => ({
     marginTop: 6,
   },
   noPosts: {
-    marginLeft: "23px",
+    marginLeft: [isEAForum ? 0 : "23px"],
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsTimeBlock.tsx:51]: @agnesstenlund you don't need those square brackets

```suggestion
    marginLeft: isEAForum ? 0 : 23,
```

(Also whoever wrote that file originally didn't realize that raw numbers get interpreted as pixels.

Comment (agnesstenlund) [reply packages/lesswrong/components/posts/PostsTimeBlock.tsx:51]: thank you 🤗

File: packages/lesswrong/lib/collections/posts/sortOrderOptions.tsx [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
-  magic: { label: 'Magic (New & Upvoted)', tooltip: 'Posts with the highest karma from the past few days' },
+  magic: {
+    label: isEAForum ? 'New & upvoted' : 'Magic (New & Upvoted)',
+    tooltip: 'Posts with the highest karma from the past few days',
+  },
   top: { label: 'Top' },
-  topAdjusted: { label: 'Top (Inflation Adjusted)', tooltip: 'Posts with the highest karma relative to those posted around the same time' },
-  recentComments: { label: 'Recent Comments' },
+  topAdjusted: {
+    label: isEAForum ? 'Top (Inflation adjusted)' : 'Top (Inflation Adjusted)',
```
Comment (s-cheng) [inline packages/lesswrong/lib/collections/posts/sortOrderOptions.tsx:15]: I think I prefer `Top (inflation adjusted)`

File: packages/lesswrong/components/posts/PostsPage/ContentType.tsx [hunk 3]
```diff
@@ -29,6 +29,10 @@ const styles = (theme: ThemeType): JssStyles => ({
   tooltipTitle: {
     marginBottom: 8,
   },
+  sectionTitle: {
+    fontSize: 14,
+    marginBottom: -4,
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsPage/ContentType.tsx:34]: Can we try to avoid negative margins? I think you can accomplish the same thing by making `marginBottom: 4` and setting the `noBottomPadding` option on `<SectionTitle>`?

File: packages/lesswrong/components/posts/AllPostsPage.tsx [hunk 4]
```diff
@@ -1,23 +1,37 @@
...
+    ? {
+      border: "none",
+      borderTop: `1px solid ${theme.palette.grey[300]}`,
+    }
+    : {
+      display: "none",
+    },
 });
 
 export const timeframes = {
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AllPostsPage.tsx:37]: Prob want to sentence case `All time` here

File: packages/lesswrong/components/common/CommentsListCondensed.tsx [hunk 5]
```diff
@@ -55,7 +55,7 @@ const CommentsListCondensed = ({label, terms, initialLimit, itemsPerPage, showTo
 const CommentsListCondensedComponent = registerComponent(
   'CommentsListCondensed',
   CommentsListCondensed,
-  {styles}
+  {styles, stylePriority: 1},
```
Comment (s-cheng) [inline packages/lesswrong/components/common/CommentsListCondensed.tsx:58]: This doesn't look like it's used on the All Posts page, why this change?

**Review by agnesstenlund — APPROVED** [review]

File: packages/lesswrong/components/posts/AllPostsList.tsx
Comment (s-cheng) [inline packages/lesswrong/components/posts/AllPostsList.tsx:1]: Just for future reference, it would be nice to get a quick PR comment here that says "The code in this file was all pulled out of `AllPostsPage.tsx`, so you don't need to read it". (I skimmed it and as far as I can tell there was just some minor code cleanup, no functional changes)

File: packages/lesswrong/components/posts/AllPostsPage.tsx [hunk 6]
```diff
@@ -28,166 +42,96 @@ export const timeframes = {
...
+            title={`${showSettings ? "Hide": "Show"} options for sorting and filtering`}
+            placement="top-end"
+          >
+            <div className={classes.title} onClick={toggleSettings}>
+              <SectionTitle title={preferredHeadingCase("All Posts")}>
+                <SortButton label={formatSort(currentSorting)} />
+              </SectionTitle>
+            </div>
+          </Tooltip>
+          {!showSettings && <hr className={classes.divider} />}
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AllPostsPage.tsx:102]: I feel like it's better to forum-gate this element rather than forum-gate the styles with a `display: none`, since then we're still including an extra element

File: packages/lesswrong/components/posts/AllPostsPage.tsx [hunk 7]
```diff
@@ -28,166 +42,96 @@ export const timeframes = {
...
+  );
 }
 
 const AllPostsPageComponent = registerComponent(
-  'AllPostsPage', AllPostsPage, {
-    styles,
-    hocs: [
-      withLocation, withUser, withTimezone,
-      withUpdateCurrentUser, withTracking
-    ]
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AllPostsPage.tsx:189]: Nice

**Review by s-cheng — APPROVED** [review]
Ok I see this was already merged, but would appreciate you addressing my comments before the UI redesign gets deployed @oetherington

File: packages/lesswrong/components/common/CommentsListCondensed.tsx (hunk 5 above)
Comment (oetherington) [reply packages/lesswrong/components/common/CommentsListCondensed.tsx:58]: A different PR changed the stylePriority of SectionTitle, which broke this.

File: packages/lesswrong/components/posts/PostsPage/ContentType.tsx (hunk 3 above)
Comment (oetherington) [reply packages/lesswrong/components/posts/PostsPage/ContentType.tsx:34]: Makes sense here so I've gone ahead and made the change. I don't really see the argument against negative margins in the general case though?

**Comment by oetherington** [comment]
@s-cheng Ah, sorry. I had 4 PRs to merge and I thought I had all the reviews, but it turns out I only had 7/8 🤦‍♂️.

Fixes here: [https://github.com/ForumMagnum/ForumMagnum/pull/6810](https://github.com/ForumMagnum/ForumMagnum/pull/6810)
